rust/connlib/tunnel/src/dns.rs

```diff
@@ -443,18 +443,82 @@ mod pattern {
     use super::*;
     use std::{convert::Infallible, fmt, str::FromStr};
 
-    #[derive(Debug, PartialEq, Eq, Hash)]
+    #[derive(Eq)]
     pub struct Pattern {
         inner: glob::Pattern,
         original: String,
     }
 
+    impl std::hash::Hash for Pattern {
+        fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+            self.original.hash(state)
+        }
+    }
+
+    impl fmt::Debug for Pattern {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            f.debug_tuple("Pattern").field(&self.original).finish()
+        }
+    }
+
+    impl PartialEq for Pattern {
+        fn eq(&self, other: &Self) -> bool {
+            self.original == other.original
+        }
+    }
+
     impl fmt::Display for Pattern {
         fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
             self.original.fmt(f)
         }
     }
 
+    impl PartialOrd for Pattern {
+        fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+            Some(self.cmp(other))
+        }
+    }
+
+    impl Ord for Pattern {
+        fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+            let mut self_rev = self.original.chars().rev();
+            let mut other_rev = other.original.chars().rev();
```

Comment:
```suggestion
            // Iterate over characters in reverse order so that e.g. `*.example.com` and `subdomain.example.com` will compare the `example.com` suffix
            let mut self_rev = self.original.chars().rev();
            let mut other_rev = other.original.chars().rev();
```